src/main/scala/io/flow/proxy/RegistryApplicationCache.scala

```
@@ -17,14 +18,19 @@ private[proxy] case class RegistryApplicationCache(client: RegistryClient)(impli
     * Get the port for the specified service, caching the
     * result. Throw an error if there is no port.
     *
-    * @param name Application name in registy
+    * @param name Application name in registry
     */
   def get(name: String): Option[Application] = {
-    cache.get(name).getOrElse {
-      fetch(name)
-    }
+    cache.getOrElse(
+      name, {
+        //attempt to load/reload cache and 'get' again, otherwise, fail
+        loadCache(100, 0)
```

I know that this solves the edge of an app added after cache was loaded; but this leads to unpredictable behavior. Better to fail fast - and in this case because get(name) already returns an option, safer would be:

```
  /**
    * Get the application with the specified name from the registry.
    *
    * @param name Application name in registry
    */
  def get(name: String): Option[Application] = {
    cache.get(name)
  }
```